app/spaces/[spaceid]/trustful/utils/useEnsData.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from 'react';
+
+import { EthereumAddress } from '../utils/types';
+// import { publicClientMainnet } from '@/lib/wallet/client'; //CHANGED publicClientMainet to client
+import { client } from '@/context/WalletContext';
+
+export enum ENSAvatarQueryStatus {
+  LOADING,
+  SUCCESS,
+  ERROR,
+}
+
+interface Props {
+  ensAddress: EthereumAddress | null;
+}
+
+export const useEnsData = ({ ensAddress }: Props) => {
+  const [primaryName, setPrimaryName] = useState<string | null | undefined>(
+    undefined,
+  );
+  const [avatarQueryStatus, setAvatarQueryStatus] =
+    useState<ENSAvatarQueryStatus>(ENSAvatarQueryStatus.LOADING);
+
+  const getEnsName = async () => {
+    if (!ensAddress) {
+      return;
+    }
+
+    const name = await client.getEnsName({
+      address: ensAddress?.address,
+    });
+
+    setPrimaryName(name);
+    setAvatarQueryStatus(ENSAvatarQueryStatus.SUCCESS);
```

Comment:
This is not making any relation with an avatar loading. Avatar is different from primaryName. No avatar is being loaded here so there is no need to store its query status and share.

Reply:
Yes.  Also, we are not using the useEnsData file. Already removed 🇹🇩